### DISCUSSION TITLE
How can I access a laragon wordpress from another computer on my local network?

### DISCUSSION THREAD

**Discussion by cmwxyz** [leokhoa/laragon#540] [Q&A]
If I navigate to the Laragon IP, I just get the Laragon landing page on the other computer. How do I get to the actual website?

**Comment by LebToki** [comment]
Hi,
use NGrok for this (you can even write a batch file to do the same upon startup if needed or whenever pressed. Mind you that free plans are timed sessions though and have changeable DNS (so you might need to share a different address on other days or instances with your coworkers).

**Connect your agent to your ngrok account**
Now that the ngrok agent is installed, let's connect it to your ngrok Account. 
- If you haven't already, [sign up (or log in)](https://dashboard.ngrok.com/) to the ngrok Dashboard and get your [Authtoken](https://dashboard.ngrok.com/get-started/your-authtoken). 

- The ngrok agent uses the authtoken (sometimes called tunnel credential) to log into your account when you start a tunnel.

Copy the value and run this command to add the authtoken in your terminal.
use the following command for this
`ngrok http --host-header=myapp.local 80
`
Enjoy!

**Reply by wiklapandu** [reply]
hello @LebToki ,
I have tried as you told above. but it became like this, the css style in my wordpress is not detected. do you know how I should do to solve it?
![image (4)](https://github.com/leokhoa/laragon/assets/50991164/82fb85f5-4836-45ac-9bf6-0f4532de2ea4)

**Reply by LebToki** [reply]
Clear your sessions completely and if everything fails and you are using this internally you can change the url of your WordPress installation (General Tab) with the value you have for your Ngrok session (temporary)

**Reply by wiklapandu** [reply]
i've try to add this code to `wp-config.php`
```php
if (!defined('WP_CLI')) {
    // define('WP_SITEURL', $_SERVER['REQUEST_SCHEME'] . '://' . $_SERVER['HTTP_HOST']);
    define('WP_SITEURL', $_SERVER['REQUEST_SCHEME'] . '://' . '3b31-116-*****.ngrok-free.app');
    define('WP_HOME',    $_SERVER['REQUEST_SCHEME'] . '://' . $_SERVER['HTTP_HOST']);
}
```
this code is working but when im going to my-account page the url is not change to ngrok url, do you know another option to change it without change the url wordpress in my databases
![image (5)](https://github.com/leokhoa/laragon/assets/50991164/dc60ec59-af0c-4977-92e7-df08fbd74d01)

**Reply by LebToki** [reply]
Have you cleared Cache within Wordpress itself ?
use the smallest caching plugin such as SpeedyCache (free and its amazing too as it does the job)

**Reply by LebToki** [reply]
besides have you tried the below as i stated earlier ??
Make sure that the Ngrok session is up and running 
instead of using the url from Ngrok (in the session)

use your local development domain

`ngrok http --host-header=*****.local 80 `

**Reply by koeliga** [reply]
I'm still having trouble with the same problem when using share function in laragon. The website loads without the css and images.
When I write `ngrok http --host-header=*****.local 80 ` it links to the laragon local host instead and if I add /mysite to the url, then along with the css also the local images dont load. Do you have any solutions? I also tried changing wp-config.php like in the thread but that didnt work either.

**Reply by LebToki** [reply]
Wordpress needs to load the address with relative URL
https://wordpress.org/plugins/absolute-relative-urls/
download this plugin try it again and hopefully it will work for you
remember to star :)

**Reply by koeliga** [reply]
Thanks for the quick reply.
The plugin you linked didn't seem to work for me but I tried a different relative url plugin and that did work for me.
https://wordpress.org/plugins/relative-url/
So thanks for the solution, I will definitely star :)

**Comment by m-pokrovskii** [comment]
Same issue. ngrok http --host-header=myapp.local 80 just redirecting me on myapp.local

**Comment by DuncanJP** [comment]
### VIEWING ANY LARAGON VIRTUAL HOST FROM ANDROID 
(ON THE SAME HOME NETWORK)
### Use the free **personalDNSfilter** App from the Google Play store.

So, not strictly an answer to accessing from a _computer_ on the same network, but if you're developing a site and want to test access from a real mobile, Android device this is the way. It also doesn't require ngrok. :-)

Tested 11th May 2024.

(Screenshots can be found after the steps listed below)

1) Open Laragon and find the IP number your Windows machine / Laragon host is using e.g. 192.168.8.103
2) Open Google Play and search for the **personalDNSfilter** App by Ingo Zenz, and install it. Again, this is App is free.
3) Once installed, start the App. Agree to the message requesting the App be allowed to access DNS traffic of your device (I chose "always allow")
4) Now we need to add our Laragon hosts. 
- Click on **Advanced Setting**s at the top of the App's main screen
- Click on **"Configure additional hosts"**. The lower half of the screen shows you the additional hosts text file contents. Add your Laragon domain(s) here but prefix them with a forward arrow '>' and follow with the IP number of your Laragon PC. For example my Laragon domain is dev.lan, and Laragon is running on 192.168.8.103 so I would enter
`>dev.lan  192.168.8.103`
- Oddly, there isn't a save button as your changes are automatically saved to the file, so just click on **Configure additional hosts** at the top of the screen to return to the App's Advanced Settings screen.
- Click on  **Advanced Settings** to return to the App's main screen.
- Click the **Restart** button, which will read the config file we've just updated with our Laragon hosts. 
5) On your Android phone open Chrome, for example, and in the address field enter your Laragon host prefixed with https:// if you're using that or http:// otherwise (my domain is **https://dev.lan** in this example)
- Chrome displays a security warning when an SSL certificate is self-signed, such as Laragon uses. Click on Advanced and agree to proceed to your Laragon site.
- Now if your Windows Firewall has been configured to allow port 443 (SSL) on your private network, voila! Your Laragon host will appear on Android! If not, yes, go play with the firewall and allow Apache, Nginx port 443 private access.

That's it!  
Don't alter any other App setting or change anything that was already in the hosts config file.

### Disclaimer
I haven't tested this method beyond what's described above, but it seems to work perfectly.
Tethering a device to the Android phone's network using this mechanism alas doesn't give that device access to Laragon's hosts, only the Android phone itself. 

### Screenshots
![personalDNSfilter_a](https://github.com/leokhoa/laragon/assets/44913277/827ce9bd-b5ad-4f22-adab-d5440c82a397)

![personalDNSfilter_b](https://github.com/leokhoa/laragon/assets/44913277/2e011097-4b4c-4b9b-bdd2-00d3748b41c7)

**Comment by DuncanJP** [comment]
### VIEWING ANY LARAGON VIRTUAL HOST FROM IOS
(ON THE SAME HOME NETWORK)

**Use the free DNSCloak App**
Again, not strictly answering the question of access from a _computer_ but hopefully close enough to be relevant and helpful.

Similar to the Android access solution, you add the Laragon hosts to a text config file which is read by an App which alters DNS for the IOS device. No jailbreaking or anything bad is required, nothing but a free App and a test file you create on your IOS device and store anywhere.

**Step 1: Prepare your ‘cloak’ file.** 
On IOS, open the Notes app and type in the entry for the Laragon host/hosts required, each on a single line:. For example, the single Laragon host which I connect to via https://dev.lan on my local network:
`dev.lan 192.168.8.103`

- To save this file, click the triple dot, then click the Square+Up-arrow symbol to _share_ the file. Change from the default “Collaborate” to “Send copy” and “Save to Files” appears. 
- Choose Location “On My iPad” … “Microsoft Office 365” folder (in my case) … and give file the name “dev”. It will append “.txt”.
- The file can be found again with the Files App.

**Step 2: Download the free [DNSCloak ](https://apps.apple.com/us/app/dnscloak-secure-dns-client/id1452162351?platform=iphone) app.** 
- Click the **top left** hamburger menu. 
- Click the last "**Advanced Options**" menu. 
- Scroll down until you see an umbrella and "Enable Cloaking". 
  - When you enable cloaking, it will ask you for the file you prepares above. Select it. 
- Now back on the main menu, click on any of the DNS servers listed e.g. Adguard-dns (always a good one). The IOS device will now resolve the hostname as long as it's on the same local (home) network as the Laragon PC.

### Disclaimer
This was last tested 01.01.2023 on both iPhone and iPad. I'm not greatly familiar with IOS and the steps may need to be tweaked as the Apps change with more recent updates, but this allows testing of your Laragon hosts on real Safari browsers, for instance, on IOS devices.
